### PR TITLE
docs - new 6x guc enable_implicit_timeformat_YYYYMMDDHH24MISS

### PR DESCRIPTION
**Pull request by lisakowen**
doc the new guc enable_implicit_timeformat_YYYYMMDDHH24MISS.  in this PR:
- add the new guc to the guc reference
- update to the migration topic, which mentions that greenplum no longer automatically converte from the deprecated YYYYMMDDHH24MISS timestamp format.  (the existing examples don't look that useful to me, but i assume that they are there for a reason, so kept them.)

### CODE REVIEW

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 1]
```diff
@@ -1835,6 +1838,38 @@
...
+              <entry colname="col1">Value Range</entry>
+              <entry colname="col2">Default</entry>
+              <entry colname="col3">Set Classifications</entry>
+            </row>
+          </thead>
+          <tbody>
+            <row>
+              <entry colname="col1">Boolean</entry>
+              <entry colname="col2">off</entry>
+              <entry colname="col3">master<p>session</p><p>reload</p></entry>
```
Comment (ashwinstar) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml:1866]: I keep forgetting what does `master` mean here? This guc ideally is set in sync cluster wide.

File: gpdb-doc/dita/install_guide/migrate.xml [hunk 2]
```diff
@@ -238,20 +238,35 @@
           previous Greenplum Database releases. You can still specify the
             <codeph>YYYYMMDDHH24MISS</codeph> format in conversion functions such as
             <codeph>to_timestamp</codeph> and <codeph>to_char</codeph> for compatibility with other
-          database systems. You can use input formats for converting text to date or timestamp
-          values to avoid unexpected results or query execution failures. For example, this
+          database systems. You have two options to avoid unexpected results or
+          query execution failures:
+          <ul>
+            <li>Set the <codeph><xref href="../ref_guide/config_params/guc-list.xml#enable_implicit_timeformat_YYYYMMDDHH24MISS"
```
[the remark this answers is not in the document]
Comment (lisakowen) [reply gpdb-doc/dita/install_guide/migrate.xml]: i moved the guc after the input format conversion option, noted that the input format conversion is the recommended option, and added a note after the list to the same affect.

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml (hunk 1 above)
Comment (lisakowen) [reply gpdb-doc/dita/ref_guide/config_params/guc-list.xml:1866]: the master set classification indicates that the guc value is passed to the segments at runtime (see https://greenplum.docs.pivotal.io/6-18/ref_guide/config_params/guc_config.html#topic_cyz_p2l_z4).  

i used the same set classifications for this guc as was currently identified for gp_use_legacy_hashops - the guc definition for both includes PGC_USERSET and COMPAT_OPTIONS_PREVIOUS.

**Review by dyozie — APPROVED** [review]

File: gpdb-doc/dita/ref_guide/config_params/guc-list.xml [hunk 3]
```diff
@@ -1835,6 +1838,38 @@
       </table>
     </body>
   </topic>
+  <topic id="enable_implicit_timeformat_YYYYMMDDHH24MISS">
+    <title>enable_implicit_timeformat_YYYYMMDDHH24MISS</title>
+    <body>
+      <p>Enables or disables implicit conversion of a string with the deprecated
+        <i>YYYYMMDDHH24MISS</i> timestamp format to a valid date/time type.</p>
```
Comment (dgkimura) [inline gpdb-doc/dita/ref_guide/config_params/guc-list.xml:1845]: Nit: this sounds to me like the timestamp format is deprecated, but really it's just the implicit conversion that is deprecated.  Maybe it should be "Enables or disables the deprecated implicit conversion..."

**Review by dgkimura — APPROVED** [review]
Looks good to me.  Small hit in word ordering.  Feel free to address or ignore as you see fit.

**Pull request merged**